plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -163,6 +165,7 @@ public CompletableFuture<ConnectorSplitBatch> getNextBatch(ConnectorPartitionHan
                 finish();
                 return completedFuture(NO_MORE_SPLITS_BATCH);
             }
+            pathDomain = getPathDomain(effectivePredicate);
```

Comment:
You should be able to calculate this just once in the constructor based on the table handle's enforcedPredicate

Reply:
Moved to the constructor and also fixed `applyFilter()`.